source/Octopus.Server.Client/Repositories/Async/ProjectRepository.cs

```diff
@@ -99,89 +153,188 @@ public Task<GitCommitResource> GetGitCommit(ProjectResource projectResource, str
-            return Client.ListAll<ChannelResource>(project.Link("Channels"));
+            return GetAllChannels(project, CancellationToken.None);
+        }
+
+        public Task<IReadOnlyList<ChannelResource>> GetAllChannels(ProjectResource project, CancellationToken cancellationToken)
+        {
+            return Client.ListAll<ChannelResource>(project.Link("Channels"), cancellationToken);
         }
 
         public Task<ProgressionResource> GetProgression(ProjectResource project)
         {
-            return Client.Get<ProgressionResource>(project.Link("Progression"));
+            return GetProgression(project, CancellationToken.None);
+        }
+
+        public Task<ProgressionResource> GetProgression(ProjectResource project, CancellationToken cancellationToken)
+        {
+            return Client.Get<ProgressionResource>(project.Link("Progression"), cancellationToken);
         }
 
         public Task<ResourceCollection<ProjectTriggerResource>> GetTriggers(ProjectResource project)
         {
-            return Client.List<ProjectTriggerResource>(project.Link("Triggers"));
+            return GetTriggers(project, CancellationToken.None);
+        }
+
+        public Task<ResourceCollection<ProjectTriggerResource>> GetTriggers(ProjectResource project, CancellationToken cancellationToken)
+        {
+            return Client.List<ProjectTriggerResource>(project.Link("Triggers"), cancellationToken);
         }
 
         public Task<IReadOnlyList<ProjectTriggerResource>> GetAllTriggers(ProjectResource project)
         {
-            return Client.ListAll<ProjectTriggerResource>(project.Link("Triggers"));
+            return GetAllTriggers(project, CancellationToken.None);
+        }
+
+        public Task<IReadOnlyList<ProjectTriggerResource>> GetAllTriggers(ProjectResource project, CancellationToken cancellationToken)
+        {
+            return Client.ListAll<ProjectTriggerResource>(project.Link("Triggers"), cancellationToken);
         }
 
         public Task SetLogo(ProjectResource project, string fileName, Stream contents)
         {
-            return Client.Post(project.Link("Logo"), new FileUpload { Contents = contents, FileName = fileName }, false);
+            return SetLogo(project, fileName, contents, CancellationToken.None);
+        }
+
+        public Task SetLogo(ProjectResource project, string fileName, Stream contents, CancellationToken cancellationToken)
+        {
+            return Client.Post(project.Link("Logo"), new FileUpload { Contents = contents, FileName = fileName }, false, cancellationToken);
         }
 
         public Task<ProjectEditor> CreateOrModify(string name, ProjectGroupResource projectGroup, LifecycleResource lifecycle)
         {
-            return new ProjectEditor(this, new ChannelRepository(Repository), new DeploymentProcessRepository(Repository), new ProjectTriggerRepository(Repository), new VariableSetRepository(Repository)).CreateOrModify(name, projectGroup, lifecycle);
```

Comment:
All `CreateOrModify` call is directed to `CreateOrModify(string, ProjectGroupResource, LifecycleResource, string, CancellationToken)` - line 277 below